src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -129,6 +134,15 @@ struct HipEventProfiler
 };
 #endif
 
+inline bool ShouldAllocateWorkSpaceBufferForWRW(const miopen::conv::ProblemDescription& problem)
+{
+    return problem.GetSpatialDims() == 3 &&
```

Comment:
`prob.Is3d()`? 
Also, break down the checks into separate bools for better readability. May be add informative comments as well.

Reply:
`prob.Is3d()` since ck is currently only supporting 3d kernel.